mailmerge/template_message.py

```diff
@@ -244,11 +266,72 @@ def _transform_attachments(self):
                 'Content-Disposition',
                 'attachment; filename="{}"'.format(basename),
             )
+
+            # When processing inline images in the email body, we will
+            # reference the Content-ID for an attachment with the same path
+            # using 'cid:[content-id]'.
+            cid_header = self._make_attachment_content_id_header(path)
+            part.add_header('Content-Id', cid_header)
+
             self._message.attach(part)
 
         # Remove the attachment header, it's non-standard for email
         del self._message['attachment']
 
+    def _transform_attachment_references(self):
+        """
+        Replace references to inline-images in the email body's HTML content.
+
+        Specifically, match inline-image src attributes with content-ids from
+        image attachments, if available.
+        """
+        if not self._message.is_multipart():
+            return
+
+        for part in self._message.walk():
+            if not part['Content-Type'].startswith('text/html'):
+                continue
+
+            html = part.get_payload(decode=True).decode('utf-8')
+            document = html5lib.parse(html, namespaceHTMLElements=False)
```

Comment:
What happens if the HTML fails to parse?

Reply:
According to the [`html5lib` docs](https://html5lib.readthedocs.io/en/latest/#usage), the parser does not raise an exception by default. Do you think we should enable `strict` mode to help users prevent sending emails will ill-formatted HTML?

Reply:
Ah, excellent.  I think this is good.